src/plugins/teleop/Teleop.cc

```diff
@@ -0,0 +1,272 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <iostream>
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif
+#include <ignition/msgs/twist.pb.h>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+#include "Teleop.hh"
+
+#include <string>
+
+#include <ignition/plugin/Register.hh>
+
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/MainWindow.hh>
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  enum class keyLinear{
+    kforward,
+    kbackward,
+    kstop,
+  };
+
+  enum class keyAngular{
+    kleft,
+    kright,
+    kstop,
+  };
+
+  class TeleopPrivate
+  {
+    /// \brief Node for communication.
+    public: ignition::transport::Node node;
+
+    /// \brief Topic. Set '/cmd_vel' as default.
+    public: std::string topic = "/cmd_vel";
+
+    /// \brief Publisher.
+    public: ignition::transport::Node::Publisher cmdVelPub;
+
+    /// \brief Linear velocity.
+    public: double linearVel = 0;
+    /// \brief Angular velocity.
+    public: double angularVel = 0;
+
+    /// \brief Linear direction.
+    public: int linearDir = 0;
+    /// \brief Angular direction.
+    public: int angularDir = 0;
+
+    /// \brief Linear state setted by keyboard input.
+    public: keyLinear linearState = keyLinear::kstop;
+    /// \brief Angular state setted by keyboard input.
+    public: keyAngular angularState = keyAngular::kstop;
+
+    public: bool keyEnable = false;
+    public: bool newTopic = true;
+
+  };
+}
+}
+}
+
+using namespace ignition;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+Teleop::Teleop(): Plugin(), dataPtr(new TeleopPrivate)
+{
+}
+
+/////////////////////////////////////////////////
+Teleop::~Teleop()
+{
+}
+
+/////////////////////////////////////////////////
+void Teleop::LoadConfig(const tinyxml2::XMLElement *_pluginElement)
+{
+  if (this->title.empty())
+    this->title = "Teleop";
+
+  ignition::gui::App()->findChild
+    <ignition::gui::MainWindow *>()->QuickWindow()->installEventFilter(this);
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnTeleopTwist()
+{
+  ignition::msgs::Twist cmdVelMsg;
+
+  cmdVelMsg.mutable_linear()->set_x(
+      this->dataPtr->linearDir * this->dataPtr->linearVel);
+  cmdVelMsg.mutable_angular()->set_z(
+      this->dataPtr->angularDir * this->dataPtr->angularVel);
+
+  if (this->dataPtr->newTopic){
+    this->dataPtr->cmdVelPub = ignition::transport::Node::Publisher();
+    this->dataPtr->cmdVelPub =
+    this->dataPtr->node.Advertise<ignition::msgs::Twist>(this->dataPtr->topic);
```

Comment:
Style nit: indent line continuations

```suggestion
        this->dataPtr->node.Advertise<ignition::msgs::Twist>(this->dataPtr->topic);
```

Reply:
Done